pkg/influx/recorder_test.go

```diff
@@ -0,0 +1,91 @@
+package influx
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRecorder(t *testing.T) {
+	reg := prometheus.NewRegistry()
+	rec := NewRecorder(reg)
+
+	tests := map[string]struct {
+		measure        func(r Recorder)
+		expMetricNames []string
+		expMetrics     string
+	}{
+		"Measure incoming points": {
+			measure: func(r Recorder) {
+				r.measureMetricsParsed(3)
+			},
+			expMetricNames: []string{
+				"influxdb_proxy_ingester_metrics_parsed_total",
+			},
+			expMetrics: `
+# HELP influxdb_proxy_ingester_metrics_parsed_total The total number of metrics that have been parsed.
+# TYPE influxdb_proxy_ingester_metrics_parsed_total counter
+influxdb_proxy_ingester_metrics_parsed_total 3
+`,
+		},
+		"Measure rejected samples": {
+			measure: func(r Recorder) {
+				r.measureMetricsRejected(3)
+			},
+			expMetricNames: []string{
+				"influxdb_proxy_ingester_metrics_rejected_total",
+			},
+			expMetrics: `
+# HELP influxdb_proxy_ingester_metrics_rejected_total The total number of metrics that were rejected.
+# TYPE influxdb_proxy_ingester_metrics_rejected_total counter
+influxdb_proxy_ingester_metrics_rejected_total 3
+`,
+		},
+		"Measure conversion duration": {
+			measure: func(r Recorder) {
+				r.measureConversionDuration(15 * time.Second)
+			},
+			expMetricNames: []string{
+				"influxdb_proxy_ingester_data_conversion_seconds",
+			},
+			expMetrics: `
+# HELP influxdb_proxy_ingester_data_conversion_seconds Time (in seconds) spent converting ingested InfluxDB data into Prometheus data.
+# TYPE influxdb_proxy_ingester_data_conversion_seconds histogram
+influxdb_proxy_ingester_data_conversion_seconds_bucket{le="0.005"} 0
```

Comment:
Do we need to set some labels? I am not sure how that works, but we want to make sure the metrics get tagged with the hostname and cluster, etc.  That's normally part of the Registerer I think?  Can you make a change in the test so a label gets applied to these metrics so we can confirm that works?